File: _fix_members.py
import pathlib
p = pathlib.Path(r'C:\xampp\htdocs\welfare\treasurer\members.php')
c = p.read_text(encoding='utf-8')

# Fix 1: executive badge column
old1 = '                                    </td>' + '\n' + '                                    <td>' + '\n' + '                                         <div class="btn-group">'
new1 = '                                    </td>' + '\n' + '                                    <td>' + '\n' + '                                        <?php echo getExecutiveBadge($member[' + "'" + 'executive_level' + "'" + '] ?? ' + "'" + 'none' + "'" + '); ?>' + '\n' + '                                    </td>' + '\n' + '                                    <td>' + '\n' + '                                         <div class="btn-group">'

print('exec found:', old1 in c)
c = c.replace(old1, new1)
print('exec done:', new1 in c)

# Fix 2: row click handler
old2 = '    // Promote to executive'
new2 = '    // Make entire member row clickable to open member detail' + '\n' + '    document.querySelectorAll("#membersTable tbody tr").forEach(function(row) {' + '\n' + '        row.style.cursor = "pointer";' + '\n' + '        row.addEventListener("click", function(e) {' + '\n' + '            if (e.target.closest("button, a, input, select, textarea")) {' + '\n' + '                return;' + '\n' + '            }' + '\n' + '            var memberId = row.getAttribute("data-member-id");' + '\n' + '            if (memberId) {' + '\n' + '                window.location.href = "<?php echo APP_URL; ?>/treasurer/member_detail.php?member_id=" + encodeURIComponent(memberId);' + '\n' + '            }' + '\n' + '        });' + '\n' + '    });' + '\n\n' + '    // Promote to executive'

print('js found:', old2 in c)
c = c.replace(old2, new2)
print('js done:', new2 in c)

p.write_text(c, encoding='utf-8')
print('DONE')

File: _fix_members.php
<?php
$file = 'C:\xampp\htdocs\welfare\treasurer\members.php';
$content = file_get_contents($file);

// Fix 1: Add executive badge column
$search1 = "                                    </td>" . "\n" . "                                    <td>" . "\n" . "                                         <div class=\"btn-group\">";
$replace1 = "                                    </td>" . "\n" . "                                    <td>" . "\n" . "                                        <?php echo getExecutiveBadge(\$member['executive_level'] ?? 'none'); ?>" . "\n" . "                                    </td>" . "\n" . "                                    <td>" . "\n" . "                                         <div class=\"btn-group\">";

if (strpos($content, $search1) !== false) {
    $content = str_replace($search1, $replace1, $content);
    echo "Added executive badge column" . "\n";
} else {
    echo "WARNING: Could not find executive column insertion point" . "\n";
}

// Fix 2: Add row click handler
$search2 = "    // Promote to executive";
$replace2 = "    // Make entire member row clickable to open member detail" . "\n" . "    document.querySelectorAll('#membersTable tbody tr').forEach(function(row) {" . "\n" . "        row.style.cursor = 'pointer';" . "\n" . "        row.addEventListener('click', function(e) {" . "\n" . "            if (e.target.closest('button, a, input, select, textarea')) {" . "\n" . "                return;" . "\n" . "            }" . "\n" . "            var memberId = row.getAttribute('data-member-id');" . "\n" . "            if (memberId) {" . "\n" . "                window.location.href = '<?php echo APP_URL; ?>/treasurer/member_detail.php?member_id=' + encodeURIComponent(memberId);" . "\n" . "            }" . "\n" . "        });" . "\n" . "    });" . "\n\n" . "    // Promote to executive";

if (strpos($content, $search2) !== false) {
    $content = str_replace($search2, $replace2, $content);
    echo "Added row click navigation" . "\n";
} else {
    echo "WARNING: Could not find promote executive comment" . "\n";
}

file_put_contents($file, $content);
echo "Done" . "\n";
?>
